From 8af24a4f07f8cf8c955075948c9073696144cc8d Mon Sep 17 00:00:00 2001
From: Michael Shamis <michaelsh@marvell.com>
Date: Thu, 14 Jan 2021 15:56:48 +0200
Subject: [PATCH 1154/1921] arm64: defconfig: enable crypto algorithms

Enable CONFIG_CRYPTO_MD5, CONFIG_CRYPTO_DES,
       CONFIG_CRYPTO_AUTHENC, CONFIG_CRYPTO_SHA512
Enabling these algorithms allows to use
crypto_safexcel.ko kernel object

Signed-off-by: Michael Shamis <michaelsh@marvell.com>
Change-Id: I9896c681bf53ab45106b699ca521b55e9f705035
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/43909
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/configs/marvell_v8_sdk_defconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/configs/marvell_v8_sdk_defconfig b/arch/arm64/configs/marvell_v8_sdk_defconfig
index 26492b7d85d9..dcb51ef354e8 100644
--- a/arch/arm64/configs/marvell_v8_sdk_defconfig
+++ b/arch/arm64/configs/marvell_v8_sdk_defconfig
@@ -834,7 +834,11 @@ CONFIG_9P_FS=y
 CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ISO8859_1=y
 CONFIG_SECURITY=y
+CONFIG_CRYPTO_AUTHENC=y
 CONFIG_CRYPTO_ECHAINIV=y
+CONFIG_CRYPTO_MD5=y
+CONFIG_CRYPTO_SHA512=y
+CONFIG_CRYPTO_DES=y
 CONFIG_CRYPTO_ANSI_CPRNG=y
 CONFIG_CRYPTO_DEV_SAFEXCEL=m
 CONFIG_CMA_SIZE_MBYTES=256
-- 
2.31.1

